### DISCUSSION TITLE
scvelo.pp.filter_and_normalize n_top_genes

### DISCUSSION THREAD

**Discussion by xzhang08** [theislab/scvelo#880] [Q&A] [answered]
Hi @WeilerP, great tool for trajectory inference!!!

I have a question about n_top_genes in scvelo.pp.filter_and_normalize. Is the n_top_genes chosen based on matrix X or unspliced/spliced?

Thanks!!

**Comment by WeilerP** [comment, marked as answer]
@xzhang08, it's chosen based on `adata.X`. See [here](https://github.com/theislab/scvelo/blob/85295fc43e9582423e4b977d5fba5c2c65869d71/scvelo/preprocessing/utils.py#L397).

**Reply by xzhang08** [reply]
Thanks so much for your fast reply. I have one more question if you don't mind. I got opposite directions for certain areas on the stream plot using different number of top genes. So the question is which results I should rely on? Is there a way to evaluate the best number of top genes for a particular dataset?
